src/main/src/main/java/org/geoserver/ows/LanguageURLMangler.java

```diff
@@ -20,24 +19,23 @@ public void mangleURL(
             StringBuilder baseURL, StringBuilder path, Map<String, String> kvp, URLType type) {
         if (Dispatcher.REQUEST.get() == null || !type.equals(URLType.SERVICE)) return;
 
-        String languageParameter = null;
-        String acceptLanguagesParameter = null;
+        final Map<String, Object> rawKvp =
+                Optional.ofNullable(Dispatcher.REQUEST.get().rawKvp).orElse(new HashMap<>());
 
-        HttpServletRequest httpRequest = Dispatcher.REQUEST.get().getHttpRequest();
-        Enumeration<String> parameterNames = httpRequest.getParameterNames();
+        final Optional<String> languageParameter =
+                rawKvp.keySet().stream()
+                        .filter(param -> param.equalsIgnoreCase(LANGUAGE))
+                        .findFirst();
```

Comment:
the raw kvp map is an HasMap implementation that is already ignore cased https://github.com/geoserver/geoserver/blob/9e40cbbccb9626b8d08516c8eacaf5a8f168189b/src/ows/src/main/java/org/geoserver/ows/util/KvpMap.java#L19. We can directly use the contains method of it without iterating before.